mate-panel/panel-menu-bar.c

```diff
@@ -184,6 +198,11 @@ static void panel_menu_bar_init(PanelMenuBar* menubar)
 
 	gtk_menu_item_set_submenu(GTK_MENU_ITEM(menubar->priv->applications_item), menubar->priv->applications_menu);
 	gtk_menu_shell_append(GTK_MENU_SHELL(menubar), menubar->priv->applications_item);
+#ifdef HAVE_WAYLAND
+	if (GDK_IS_WAYLAND_DISPLAY (gtk_widget_get_display (GTK_WIDGET (menubar)))) {
+		wayland_popup_menu_setup(menubar->priv->applications_menu, menubar->priv->applications_item);
```

Comment:
It is regular / normal `GtkMenuBar`... This makes me think that something is wrong...

If this is needed in panel then it would mean that it is needed in all other apps and I am sure that is not the case, is it?

Reply:
Explained below